src/python/pants/backend/python/tasks/python_execution_task_base.py

```diff
@@ -116,8 +116,17 @@ def create_pex(self, pex_info=None):
           # Add the extra requirements first, so they take precedence over any colliding version
           # in the target set's dependency closure.
           pexes = [extra_requirements_pex] + pexes
-        constraints = {constraint for rt in relevant_targets if is_python_target(rt)
-                       for constraint in PythonSetup.global_instance().compatibility_or_constraints(rt)}
+
+        unique_constraints = {
+          constraint for rt in relevant_targets if is_python_target(rt)
+          for constraint in PythonSetup.global_instance().compatibility_or_constraints(rt)
+        }
+        self.context.log.debug('unique_constraints:\n{}'.format(unique_constraints))
+
+        if pin_selected_interpreter:
```

Comment:
This flag seems to disable the `unique_constraints` codepath entirely. should those be computed under that branch of the if instead?

It seems like rather than "pinning", this is more like: "use_global_python_interpreter" or something, because what it pins to is specifically the `PythonInterpreter` product. It should definitely have a mention in the docstring.

Reply:
I've removed the extra argument to `create_pex()` in favor of removing the `unique_constraints` code path entirely, as every consumer of this method is always going to depend on requirements pexes resolved specifically for the single global selected interpreter. I've also added a docstring clarifying all of this.

Reply:
As noted in https://github.com/pantsbuild/pants/pull/7563#issuecomment-488205889, I've added back the `pin_selected_interpreter` argument, but added a descriptive docstring. I'll be following up with a change that removes this.